python/test/unit/fem/test_expression_evaluation.py

```diff
@@ -97,3 +93,66 @@ def grad_expr1(x):
     b2.interpolate(grad_expr1)
 
     assert np.isclose((b2.vector - b.vector).norm(), 0.0)
+
+
+def test_simple_evaluation():
+    """Test evaluation of UFL Expression.
+
+    This test evaluates a UFL Expression on two cells in the mesh and compares the
+    result with an analytical expression.
+
+    For a donor function f(x, y) = (x^2 + 2*y^2) result is compared with the
+    exact gradient grad f(x, y) = [2*x, 4*y]. In this test, f(x, y) is first
```

Comment:
I would put the equation a separate line - it's not so easy to parse the 'grad' with the 'f(x, y)'.